types/validator_set.go

```diff
@@ -569,22 +549,46 @@ func (vals *ValidatorSet) updateWithChangeSet(changes []*Validator, allowDeletes
 
 	// Compute the priorities for updates
 	numNewValidators := computeNewPriorities(updates, vals, updatedTotalVotingPower)
-	if len(vals.Validators)+numNewValidators <= len(deletes) {
-		err = fmt.Errorf("applying the validator changes would result in empty set")
+	if len(vals.Validators)+numNewValidators == len(deletes) {
+		err = errors.New("applying the validator changes would result in empty set")
```

Comment:
return errors.New...

Reply:
(?) fixed with https://github.com/tendermint/tendermint/pull/3301/commits/e9a297b19af466a8e7aa1482f5169b29eae82c67

Reply:
ah sorry you want the return

Reply:
Yes, I think Anton means:
```suggestion
		return errors.New("applying the validator changes would result in empty set")
```

Reply:
What @Liamsi said